java/vector/src/main/java/org/apache/arrow/vector/file/ArrowReader.java

```diff
@@ -135,13 +135,20 @@ public long bytesRead() {
 
   @Override
   public void close() throws IOException {
+    close(true);
+  }
+
+  public void close(boolean closeReadChannel) throws IOException {
```

Comment:
Add some docs?

Reply:
Yeah, good idea. Need to start properly documented the Java classes.  Added this and cleaned up the docs for the rest of the public methods